src/middlewares/patient.js

```diff
@@ -3,18 +3,26 @@
 
     var keyFor = {
         "Observation": "subject",
-        "MedicationPrescription": "patient"
+        "MedicationPrescription": "patient",
+        "Encounter": "patient"
```

Comment:
Josh, sure, we can expose two instances of the client as you outlined with and without the patient context middleware enabled. Furthermore, the client can decide which of the two client to instantiate and expose based on the launch conditions (if there is a patient context). This way, an app which expects to work in patient context will not produce incorrect results by accident.

Still, this does not resolve the need of the patient context middleware to have a list of the resources for which it should apply the patient context (unless there is something that I don't yet understand here).